packages/grid/x-data-grid/src/tests/filtering.DataGrid.test.tsx

```diff
@@ -71,30 +139,58 @@ describe('<DataGrid /> - Filter', () => {
       operatorValue?: string;
       value?: any;
       field?: string;
+      imperativeFilterModel?: boolean;
     } & Partial<Omit<DataGridProps, 'columns'>>,
   ) => {
-    const { operatorValue, value, rows, columns, field = 'brand', ...other } = props;
+    const {
+      operatorValue,
+      value,
+      rows,
+      columns,
+      field = 'brand',
+      imperativeFilterModel = true,
+      ...other
+    } = props;
     return (
       <div style={{ width: 300, height: 300 }}>
         <DataGrid
           autoHeight={isJSDOM}
           columns={columns || baselineProps.columns}
           rows={rows || baselineProps.rows}
-          filterModel={{
-            items: [
-              {
-                columnField: field,
-                value,
-                operatorValue,
-              },
-            ],
-          }}
+          filterModel={
+            imperativeFilterModel
```

Comment:
We never did that. You could check the `initialState` prop and if it's defined, don't set the model.